packages/grid/x-data-grid/src/internals/hooks/core/useGridStateInitialization.ts

```diff
@@ -73,11 +73,11 @@ export const useGridStateInitialization = <Api extends GridApiCommon>(
         // Events are here to forward to other hooks and apply changes.
         // You are trying to update several states in a no isolated way.
         throw new Error(
-          `You're not allowed to update several sub-state in one transaction. You already updated ${
-            updatedControlStateIds[0]
-          }, therefore, you're not allowed to update ${updatedControlStateIds.join(
-            ', ',
-          )} in the same transaction.`,
+          `You're not allowed to update several sub-state in one transaction. You already updated ${JSON.stringify(
```

Comment:
Why do we need this ?

Reply:
Because I get in my console 

> You're not allowed to update several sub-state in one transaction. You already updated [object Object]

Reply:
Oh right I turned it into an object some time ago.
But the `hasPropChanged` has no use here

```suggestion
        throw new Error(
          `You're not allowed to update several sub-state in one transaction. You already updated ${
            updatedControlStateIds[0].stateId
          }, therefore, you're not allowed to update ${updatedControlStateIds.join(
            ', ',
          ).map(el => el.id)} in the same transaction.`,
        );
```

Reply:
It's out of the PR topic, but why is there a `updatedControlStateIds.forEach` wherase you check juste before the length is at most 1? Would it make sense to use `console.error` or to remove the forEach?

```js
if (updatedControlStateIds.length > 1){
	throw Error(
		// ...
	)
}
// ...
updatedControlStateIds.forEach(({ stateId, hasPropChanged }) => { 
	// ...
})
```

Reply:
You are right, we could probably just run on the 1st item instead of looping